agent/api/ecr.go

```diff
@@ -22,7 +26,9 @@ type RegistryAuthenticationData struct {
 
 // ECRAuthData is the authentication details for ECR specifying the region, registryID, and possible endpoint override
 type ECRAuthData struct {
-	EndpointOverride string `json:"endpointOverride"`
-	Region           string `json:"region"`
-	RegistryID       string `json:"registryId"`
+	EndpointOverride string                         `json:"endpointOverride"`
+	Region           string                         `json:"region"`
+	RegistryID       string                         `json:"registryId"`
+	UseExecutionRole bool                           `json:"useExecutionRole"`
+	PullCredentials  credentials.IAMRoleCredentials `json:"-"`
```

Comment:
Does this need to be global scoped? Can it be `pullCredentials` instead? Advantages: 
1. No need of the json tag and not having to worry about accidentally exposing it if we screw up json marshaling in the future
2. We can add `get` and `set` methods for this field
3. We can add locks to prevent races in `get` and `set` methods.